lib/pipeline.js

```diff
@@ -497,6 +497,34 @@ class PipelineModel extends BaseModel {
     }
 
     /**
+     * Update the repository and branch
+     * @method update
+     * @return {Promise}
+     */
+    update() {
+        if (this.isDirty('scmUri')) {
+            // Lazy load factory dependency to prevent circular dependency issues
+            // https://nodejs.org/api/modules.html#modules_cycles
+            /* eslint-disable global-require */
+            const UserFactory = require('./userFactory');
+            /* eslint-enable global-require */
+
+            const userFactory = UserFactory.getInstance();
+
+            return userFactory.get({ username: Object.keys(this.admins)[0] })
+                .then(user => user.unsealToken())
+                .then(token => this.scm.decorateUrl({ scmUri: this.scmUri, token }))
+                .then((scmRepo) => {
+                    this.scmRepo = scmRepo;
+
+                    return super.update();
```

Comment:
I think they both work. I think I had that earlier. I might have accidentally crtl+z'ed too far.